app/models/manageiq/providers/openstack/event_catcher_mixin.rb

```diff
@@ -48,28 +48,30 @@ def monitor_events
     reset_event_monitor_handle
   end
 
-  def process_event(event)
-    if filtered_events.include?(event.payload[:event_type])
-      _log.info "#{log_prefix} Skipping caught event [#{event.payload["event_type"]}]"
-    else
-      _log.info "#{log_prefix} Caught event [#{event.payload["event_type"]}]"
+  def filtered?(event)
+    filtered_events.include?(event.payload[:event_type]).tap do |result|
+      _log.info "#{log_prefix} Skipping caught event [#{event.payload["event_type"]}]" if result
+    end
+  end
 
-      event_hash = {}
-      # copy content
-      content = event.payload
-      event_hash[:content] = content.reject { |k, _v| k.start_with? "_context_" }
+  def queue_event(event)
+    _log.info "#{log_prefix} Caught event [#{event.payload["event_type"]}]"
 
-      # copy context
-      event_hash[:context] = {}
-      content.select { |k, _v| k.start_with? "_context_" }.each_pair do |k, v|
-        event_hash[:context][k] = v
-      end
+    event_hash = {}
```

Comment:
I see that you are trying to stay as close as possible to the original code :) 

but I think the `event_hash` can be setup in one go
```ruby
event_hash = {
  :content => content.reject { |k, _v| k.start_with? "_context_" },
  :context => content.select { |k, _v| k.start_with? "_context_" }
  # and now the attributes
}